evennia/server/is_typing.py

```diff
@@ -0,0 +1,71 @@
+"""
+This module allows users based on a given condition (defaults to same location) to see
+whether applicable users are typing or not. Currently, only the webclient is supported.
+"""
+
+from evennia import DefaultCharacter
+from evennia.commands.default.general import CmdSay
+
+# Notification timeout in milliseconds + <=100ms polling interval in the client.
+_IS_TYPING_TIMEOUT = 1000 * 5
+
+
+def is_typing_setup(session, *args, **kwargs):
+    """
+    This fetches any aliases for the "say" command and the
+    specified notification timeout in milliseconds.
+
+    Args:
+        session: The player's current session.
+    """
+
+    options = session.protocol_flags
+    is_typing = options.get("ISTYPING", True)
+
+    if not is_typing:
+        return
+
+    session.msg(
+        is_typing={
+            "type": "setup",
+            "payload": {"say_aliases": CmdSay.aliases, "talking_timeout": _IS_TYPING_TIMEOUT},
```

Comment:
It's a dangerous practice to have core depend on default commands like this; we actively recommend users override default commands with their own alternatives, including `CmdSay`.